src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -163,10 +163,10 @@ public static string CollectionTemplate(IHtmlHelper htmlHelper)
 
         public static string DecimalTemplate(IHtmlHelper htmlHelper)
         {
-            if (htmlHelper.ViewData.TemplateInfo.FormattedModelValue == htmlHelper.ViewData.ModelMetadata.Model)
+            if (htmlHelper.ViewData.TemplateInfo.FormattedModelValue == htmlHelper.ViewData.Model)
```

Comment:
nervous about these changes now that I see you're continuing to use `ModelExplorer.Model` elsewhere.  until / unless we're confident `ViewData.Model == ModelMetadata.Model` is an invariant, stick with the one we have today.